src/lambda_handler.py

```diff
@@ -1,89 +1,207 @@
-from typing import Any, Optional, Union
+from typing import List, Optional, Tuple
+import urllib.parse
+import urllib.request
 
 # Third-Party Libraries
-import cowsay
-import cowsay.characters
-
-# cisagov Libraries
-from example import example_div
+from beanie import Document, init_beanie
+from boto3 import client as boto3_client
+from botocore.exceptions import ClientError
+from motor.motor_asyncio import AsyncIOMotorClient
 
 default_log_level = "INFO"
 logger = logging.getLogger()
 logger.setLevel(default_log_level)
 
+DEFAULT_NVD_URL = "https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-{year}.json.gz"
+NVD_FIRST_YEAR = 2002
+DEFAULT_NVD_COLLECTION = "cves"
+
+motor_client: AsyncIOMotorClient = None
+ssm_client: boto3_client = None
+
+
+class CVEDoc(Document):
+    """Python class to represent a CVE Document."""
+
+    id: str
+    cvss_score: Optional[float]
+    cvss_version: Optional[str]
+    severity: Optional[int]
+
+    class Settings:
+        """Optional settings."""
 
-def failed_task(result: dict[str, Any], error_msg: str) -> None:
-    """Update a given result because of a failure during processing."""
-    result["success"] = False
-    result["error_message"] = error_msg
+        name = DEFAULT_NVD_COLLECTION
+        validate_on_save = True
 
+    async def save(self, *args, **kwargs):
+        """Save the document to the database."""
+        # Calculate severity based on CVSS version and score.
 
-def task_default(event):
-    """Provide a result if no valid task was provided."""
-    result = {}
-    error_msg = 'Provided task "%s" is not supported.'
+        if self.cvss_version == "2.0":
+            if self.cvss_score == 10:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        elif self.cvss_version in ["3.0", "3.1"]:
+            if self.cvss_score >= 9.0:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        await super().save(*args, **kwargs)
 
-    task = event.get("task", None)
-    logging.error(error_msg, task)
-    failed_task(result, error_msg % task)
 
-    return result
+async def process_nvd(cve) -> str:
+    """Add the provided CVE to the database and return its id."""
+    # Fill fields for the CVE Documents from the given JSON files
+    cve_id = cve["cve"]["CVE_data_meta"]["ID"]
+    # Reject or remove CVEs that don't have baseMetricV2 or baseMetricV3 CVSS data
+    if not any(k in cve["impact"] for k in ["baseMetricV2", "baseMetricV3"]):
+        """If the CVE is in the database, it needs to be deleted from the database"""
 
+        if await CVEDoc.find_one(CVEDoc.id == cve_id):
+            print("x", end="")
```

Comment:
Are these `print()` lines intended to be left in?

Reply:
@jsf9k I asked the same question to @mcdonnnj. He said "That will end up in CloudTrail and will be there for confirmation/debugging still."

Reply:
@jsf9k I marked this as "resolved" so I recently unmarked it so you could officially mark it.

Reply:
@mcdonnnj - Can you comment here?  Are the `print()` lines intended to be left in?

Reply:
Pinging @mcdonnnj again here - we just want to get your confirmation on this thread.  I believe it makes sense to keep these `print()` lines in here for auditing purposes.

Reply:
@jsf9k I think it's safe to resolve this conversation- 7 weeks is too long to wait for an answer to this.